libp2p/peer/peerstore.py

```diff
@@ -10,90 +12,196 @@
+        """
+        if peer_id in self.peer_data_map:
+            return self.peer_data_map[peer_id].get_addrs()
         raise PeerStoreError("peer ID not found")
 
     def clear_addrs(self, peer_id: ID) -> None:
+        """
+        :param peer_id: peer ID to clear addrs for
+        """
         # Only clear addresses if the peer is in peer map
-        if peer_id in self.peer_map:
-            self.peer_map[peer_id].clear_addrs()
+        if peer_id in self.peer_data_map:
+            self.peer_data_map[peer_id].clear_addrs()
 
     def peers_with_addrs(self) -> List[ID]:
+        """
+        :return: all of the peer IDs which has addrs stored in peer store
+        """
         # Add all peers with addrs at least 1 to output
         output: List[ID] = []
 
-        for peer_id in self.peer_map:
-            if len(self.peer_map[peer_id].get_addrs()) >= 1:
+        for peer_id in self.peer_data_map:
+            if len(self.peer_data_map[peer_id].get_addrs()) >= 1:
                 output.append(peer_id)
         return output
 
+    def add_pubkey(self, peer_id: ID, pubkey: PublicKey) -> None:
+        """
+        :param peer_id: peer ID to add public key for
+        :param pubkey:
+        :raise PeerStoreError: if peer ID already has pubkey set
+        """
+        if peer_id in self.peer_pubkey_map:
+            raise PeerStoreError(
+                f"peer ID already has pubkey: {self.peer_pubkey_map[peer_id]}"
+            )
+        self.peer_pubkey_map[peer_id] = pubkey
+
+    def pubkey(self, peer_id: ID) -> PublicKey:
+        """
+        :param peer_id: peer ID to get public key for
+        :return: public key of the peer
+        :raise PeerStoreError: if peer ID not found
+        """
+        if peer_id in self.peer_pubkey_map:
+            return self.peer_pubkey_map[peer_id]
+        raise PeerStoreError("peer ID not found")
+
+    def add_privkey(self, peer_id: ID, privkey: PrivateKey) -> None:
+        """
+        :param peer_id: peer ID to add private key for
+        :param privkey:
+        :raise PeerStoreError: if peer ID already has privkey set
+        """
+        if peer_id in self.peer_privkey_map:
+            raise PeerStoreError(
+                f"peer ID already has privkey: {self.peer_privkey_map[peer_id]}"
+            )
+        self.peer_privkey_map[peer_id] = privkey
+
+    def privkey(self, peer_id: ID) -> PrivateKey:
+        """
+        :param peer_id: peer ID to get private key for
+        :return: private key of the peer
+        :raise PeerStoreError: if peer ID not found
+        """
+        if peer_id in self.peer_pubkey_map:
+            return self.peer_privkey_map[peer_id]
+        raise PeerStoreError("peer ID not found")
+
+    def peers_with_keys(self) -> Set[ID]:
+        """
+        :return: all of the peer IDs which has pubkey/privkey stored in peer store
+        """
+        return set(self.peer_pubkey_map.keys()).union(self.peer_privkey_map.keys())
```

Comment:
i'm curious how this is used... assuming its something in the Go impl...

Reply:
It's not actually used in the Go implementation.🤔 Removed it.